app/api/views.py

```diff
@@ -171,7 +172,9 @@ def get_queryset(self):
 
         queryset = project.documents
         if project.randomize_document_order:
-            queryset = queryset.annotate(sort_id=F('id') % self.request.user.id).order_by('sort_id')
+            random.seed(self.request.user.id)
+            value = random.randrange(2, 20)
+            queryset = queryset.annotate(sort_id=F('id') % value).order_by('sort_id', 'id')
```

Comment:
@Hironsan , really, how does it work?

Reply:
Good question.

In fact, this is not a true randomization. I tried some randomization methods and it works well on a small dataset. However, if we use a large dataset, I found the randomization takes a long time.

That's why I decided to solve only the duplication problem. There is still an issue of true randomization.

Reply:
Ok, it's not truely random, but how do these lines work? I don't understand :)

And why does this patch work? We only replace `self.request.user.id` with `f(self.request.user.id)`, where `f` isn't random function. Yeah? `value = random.randrange(2, 20)` will be the the same for the same seed (`self.request.user.id`), woudn't it be?

Reply:
@Hironsan , Ok, it's not truely random, but how do these lines work? I don't understand :)

And why does this patch work? We only replace `self.request.user.id` with `f(self.request.user.id)`, where `f` isn't random function. Yeah? `value = random.randrange(2, 20)` will be the the same for the same seed (`self.request.user.id`), woudn't it be?